gurmukhiutils/unicode.py

```diff
@@ -0,0 +1,448 @@
+        "Ú": "ਃ",  # capital u-acute letter
+        "\u02c6": "ਂ",  # circumflex accent (ˆ)
+        "\u02dc": "੍ਨ",  # small tilde (˜)
+        #
+        #
+        "<>": "ੴ",  # AnmolLipi/GurbaniAkhar variant
+        "<": "ੴ",  # GurbaniLipi variant
+        ">": "☬",  # GurbaniLipi variant
+        #
+        "Åå": "ੴ",  # AnmolLipi/GurbaniAkhar variant
+        "Å": "ੴ",  # GurbaniLipi variant
+        "å": "ੴ",  # GurbaniLipi variant
+        #
+        #
+        # AnmolLipi/GurbaniAkhar mappings:
+        "§": "੍ਹੂ",  # section symbol
+        "¤": "ੱ",  # currency symbol
+        #
+        #
+        # GurbaniLipi mappings:
+        "ç": "੍ਚ",  # c-cedilla letter
+        #
+        #
+        # AnmolLipi/GurbaniAkhar overriding GurbaniLipi mapping:
+        "Ç": "☬",  # khanda instead of california state symbol
+        #
+        #
+        # Miscellaneous:
+        "\u201a": "❁",  # single low-9 quotation (‚) mark
+        #
+        #
+        # Nullify
+        "Æ": "",  # This is either the 2nd portion of ੴ or a symbol of USA. The ੴ renders correctly according to rules above.
+        "Ø": "",  # This is a topline / shirorekha (शिरोरेखा) extender
+        "ÿ": "",  # This is the author Kulbir S Thind's stamp
+        "Œ": "",  # Box drawing left flower
+        "‰": "",  # Box drawing right flower
+        "Ó": "",  # Box drawing top flower
+        "Ô": "",  # Box drawing bottom flower
+    }
+
+    # OpenGurbaniAkhar by Sarabveer Singh (GurbaniNow)
+    special_sant_lipi_map = {
+        "Î": "\ueeec",  # replace capital i-circumflex letter, half-yayya
+        "੍ਯ": "\ueeec",  # replace unicode half-yayya
+        "î": "\ueeee",  # i-circumflex letter, open-top half-yayya
+        "ï": "\ueeef",  # i-diaeresis letter, open-top full yayya
+        "\u2080": "\uee80",  # missing subscript 0 (₀)
+        "\u2081": "\uee81",  # subscript 1 (₁) number
+        "\u2082": "\uee82",  # subscript 2 (₂) number
+        "\u2083": "\uee83",  # subscript 3 (₃) number
+        "\u2084": "\uee84",  # subscript 4 (₄) number
+        "\u2085": "\uee85",  # subscript 5 (₅) number
+        "\u2086": "\uee86",  # subscript 6 (₆) number
+        "\u2087": "\uee87",  # missing subscript 7 (₇)
+        "\u2088": "\uee88",  # subscript 8 (₈) number
+        "\u2089": "\uee89",  # missing subscript 9 (₉)
+    }
+
+    sant_lipi_to_unicode_compliant_map = {
+        "\ueeec": "੍ਯ",
+        "\ueeee": "੍ਯ",
+        "\ueeef": "ਯ",
+        # The Shabad OS Database as of 2022-06-27 only has these occur before a white-space character:
+        "\uee80": "₀",
+        "\uee81": "₁",
+        "\uee82": "₂",
+        "\uee83": "₃",
+        "\uee84": "₄",
+        "\uee85": "₅",
+        "\uee86": "₆",
+        "\uee87": "₇",
+        "\uee88": "₈",
+        "\uee89": "₉",
+    }
+
+    # Move ASCII sihari before mapping to unicode
+    ascii_base_letters = "a-zA-Z\\|^&Îîï"
+    pattern = r"(i)([" + ascii_base_letters + "])"
```

Comment:
fyi you can do
`fr"(i)([{ascii_base_letters}])"`
you generally want to avoid string concatenation because python isn't type safe, f-strings or .format are useful things for this.